pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -3218,7 +3219,7 @@ public boolean checkSubscriptionTypesEnable(SubType subType) throws Exception {
                 if (topicPolicies == null) {
                     return checkNsAndBrokerSubscriptionTypesEnable(topicName, subType);
                 } else {
-                    if (topicPolicies.getSubscriptionTypesEnabled().isEmpty()) {
+                    if (CollectionUtils.isEmpty(topicPolicies.getSubscriptionTypesEnabled())) {
```

Comment:
`topicPolicies.getSubscriptionTypesEnabled()` default value is `empty list`, when can it become `null`?
https://github.com/apache/pulsar/blob/94736a43f1b9a6d1db75936032b94eb9a11b9c0d/pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/TopicPolicies.java#L47-L48
In addition, it is recommended to combine the two judgments as below:
```
if (topicPolicies == null || CollectionUtils.isEmpty(topicPolicies.getSubscriptionTypesEnabled())) {
    return checkNsAndBrokerSubscriptionTypesEnable(topicName, subType);
} else {
    return topicPolicies.getSubscriptionTypesEnabled().contains(subType);
}
```

Reply:
topicPolicies.getSubscriptionTypesEnabled() default value is empty list.
+1

Reply:
The problem occurred when upgrading from pulsar version 2.7 to 2.8. I have added the root cause of the problem and the reproduce steps to the PR description. Please check. @yuruguo @gaozhangmin 

> In addition, it is recommended to combine the two judgments as below:
> ```
> if (topicPolicies == null || CollectionUtils.isEmpty(topicPolicies.getSubscriptionTypesEnabled())) {
>     return checkNsAndBrokerSubscriptionTypesEnable(topicName, subType);
> } else {
>     return topicPolicies.getSubscriptionTypesEnabled().contains(subType);
> }
> ```

@yuruguo I don't think we need to. The null check of topicPolicy has been done before.